src/parquet/file/metadata-internal.cc

```diff
@@ -0,0 +1,500 @@
+
+  ColumnStatistics Stats() const {
+    const format::ColumnMetaData& meta_data = column_->meta_data;
+
+    ColumnStatistics result;
+    result.null_count = meta_data.statistics.null_count;
+    result.distinct_count = meta_data.statistics.distinct_count;
+    result.max = &meta_data.statistics.max;
+    result.min = &meta_data.statistics.min;
+    return result;
+  }
+
+  Compression::type compression() const { return FromThrift(column_->meta_data.codec); }
+
+  std::vector<Encoding::type> Encodings() const {
+    const format::ColumnMetaData& meta_data = column_->meta_data;
+
+    std::vector<Encoding::type> encodings;
+    for (auto encoding : meta_data.encodings) {
+      encodings.push_back(FromThrift(encoding));
+    }
+    return encodings;
+  }
+
+  int64_t total_compressed_size() const {
+    return column_->meta_data.total_compressed_size;
+  }
+
+  int64_t total_uncompressed_size() const {
+    return column_->meta_data.total_uncompressed_size;
+  }
+
+ private:
+  const format::ColumnChunk* column_;
+};
+
+std::unique_ptr<ColumnMetaData> ColumnMetaData::GetColumnMetaData(
+    const uint8_t* metadata) {
+  return std::unique_ptr<ColumnMetaData>(new ColumnMetaData(metadata));
+}
+
+ColumnMetaData::ColumnMetaData(const uint8_t* metadata)
+    : impl_{std::unique_ptr<ColumnMetaDataImpl>(new ColumnMetaDataImpl(
+          reinterpret_cast<const format::ColumnChunk*>(metadata)))} {}
+ColumnMetaData::~ColumnMetaData() {}
+
+// column chunk
+int64_t ColumnMetaData::file_offset() const {
+  return impl_->file_offset();
+}
+
+const std::string& ColumnMetaData::file_path() const {
+  return impl_->file_path();
+}
+
+// column metadata
+Type::type ColumnMetaData::type() const {
+  return impl_->type();
+}
+
+int64_t ColumnMetaData::num_values() const {
+  return impl_->num_values();
+}
+
+std::shared_ptr<schema::ColumnPath> ColumnMetaData::path_in_schema() const {
+  return impl_->path_in_schema();
+}
+
+ColumnStatistics ColumnMetaData::Stats() const {
+  return impl_->Stats();
+}
+
+bool ColumnMetaData::is_stats_set() const {
+  return impl_->is_stats_set();
+}
+
+Compression::type ColumnMetaData::compression() const {
+  return impl_->compression();
+}
+
+std::vector<Encoding::type> ColumnMetaData::Encodings() const {
+  return impl_->Encodings();
+}
+
+int64_t ColumnMetaData::total_uncompressed_size() const {
+  return impl_->total_uncompressed_size();
+}
+
+int64_t ColumnMetaData::total_compressed_size() const {
+  return impl_->total_compressed_size();
+}
+
+// row-group metadata
+class RowGroupMetaData::RowGroupMetaDataImpl {
+ public:
+  explicit RowGroupMetaDataImpl(const format::RowGroup* row_group)
+      : row_group_(row_group) {}
+  ~RowGroupMetaDataImpl() {}
+
+  int num_columns() const { return row_group_->columns.size(); }
+
+  int64_t num_rows() const { return row_group_->num_rows; }
+
+  int64_t total_byte_size() const { return row_group_->total_byte_size; }
+
+  std::unique_ptr<ColumnMetaData> GetColumnMetaData(int i) {
```

Comment:
The problem with removing `Get` here is that the name will match the class name and it becomes a constructor. Do you have other name suggestions ?